### PR TITLE
serialize: .param - don't use ajaxSettings.traditional as a default value

### LINKED ISSUE #3023: `jQuery.param`’s behaviour differs with the one described in the docs
## docs

http://api.jquery.com/jQuery.param/

As of jQuery 1.8, the `$.param()` method no longer uses `jQuery.ajaxSettings.traditional` as its default setting and will default to `false`. For best compatibility across versions, call `$.param()` with an explicit value for the second argument and do not use defaults.

## actual

https://github.com/jquery/jquery/blob/master/src/serialize.js#L65

```javascript
	// Set traditional to true for jQuery <= 1.3.2 behavior.
	if ( traditional === undefined ) {
		traditional = jQuery.ajaxSettings && jQuery.ajaxSettings.traditional;
	}
```

### PR DESCRIPTION
**Pull request by sowcow**
(Fixes #3023)

### CODE REVIEW

**Comment by dmethvin** [comment]
This LGTM, can I land it before we do the beta? If so I want to make a note in the upgrade guide since it's a breaking change, even tho we documented it had changed as of 1.8 as #3023 mentioned.

**Comment by markelog** [comment]
> This LGTM, can I land it before we do the beta?

You mean rc? +1 from me, we should land it while we had this chance!

File: test/unit/serialize.js [hunk 1]
```diff
@@ -39,52 +37,38 @@ QUnit.test( "jQuery.param()", function( assert ) {
...
+	assert.equal( jQuery.param( params, true ), "a=0&a=1%2C2&a=3%2C4%2C5%2C6&a=%5Bobject%20Object%5D&a=17", "nested arrays (not possible when jQuery.param.traditional == true)" );
 
 	params = { a:[ 1,2 ], b:{ c:3, d:[ 4,5 ], e:{ x:[ 6 ], y:7, z:[ 8,9 ] }, f:true, g:false, h:undefined }, i:[ 10,11 ], j:true, k:false, l:[ undefined,0 ], m:"cowboy hat?" };
-	assert.equal( decodeURIComponent( jQuery.param( params, false ) ), "a[]=1&a[]=2&b[c]=3&b[d][]=4&b[d][]=5&b[e][x][]=6&b[e][y]=7&b[e][z][]=8&b[e][z][]=9&b[f]=true&b[g]=false&b[h]=&i[]=10&i[]=11&j=true&k=false&l[]=&l[]=0&m=cowboy hat?", "huge structure, forced not traditional" );
+	assert.equal( decodeURIComponent( jQuery.param( params ) ), "a[]=1&a[]=2&b[c]=3&b[d][]=4&b[d][]=5&b[e][x][]=6&b[e][y]=7&b[e][z][]=8&b[e][z][]=9&b[f]=true&b[g]=false&b[h]=&i[]=10&i[]=11&j=true&k=false&l[]=&l[]=0&m=cowboy hat?", "huge structure, forced not traditional" );
 
 	params = { "param1": null };
-	assert.equal( jQuery.param( params, false ), "param1=", "Make sure that null params aren't traversed." );
```
Comment (dmethvin) [inline test/unit/serialize.js]: It's okay to leave the explicit `false` cases in here, right?

Comment (sowcow) [inline test/unit/serialize.js]: I guess it depends on what provides more value here - explicitness, or tests that are closer to how it is expected to be used. Explicitness is related to better maintability in a way?

**Comment by dmethvin** [comment]
@sowcow will you be able to get to this soon?

**Comment by dmethvin** [comment]
Don't land this PR as it is incorporated into gh-3081 and I'll just close this PR when that one lands.

**Comment by dmethvin** [comment]
BTW, thanks for your help @sowcow! :star2:

**Pull request closed without merging**